packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,316 @@
+}
+
+interface TabsProps {
+  /**
+   * Content that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when a tab is selected.
+   */
+  onChange?: React.ReactEventHandler;
+
+  /**
+   * Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: string | null;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+
+  /**
+   * HTML Element that wraps title in Tab List.
+   */
+  as?: React.ElementType<any>;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs onChange={() => execute callback onChange}>
+  <Tab value='tab-1' tabTitle='First Tab'>Tab 1</Tab>
+  <Tab value='tab-2' tabTitle='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.onChange Callback to be executed when a Tab is selected.
+ * @param props.selected Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className Classname applied to Tabs container.
+ * @param props.as HTML Element that wraps title in Tab List.
+ */
+function Tabs({
+  children,
+  onChange,
+  selected,
+  className,
+  as = 'button',
+  ...rest
+}: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([] as Array<string>);
+  const tabListRef = useRef<HTMLDivElement>(null);
+
+  const childrenArray = React.Children.toArray(children) as Array<
+    React.ReactElement
+  >;
+
+  useEffect(() => {
+    const currentIndex = childrenArray.reduce(
+      (acc, child, index) => {
+        if (!child) {
+          return acc;
+        }
+
+        if (selected || activeTab) {
+          return [activeTab, selected].includes(child.props.value)
+            ? [...acc, index]
+            : acc;
+        }
+
+        if (child.props.default) {
+          return [...acc, index];
+        }
+
+        return acc;
+      },
+      [] as Array<number>,
+    );
+
+    setCurrentIndex(currentIndex[0]);
+  });
+
+  function handleChange(e: React.SyntheticEvent<Element, MouseEvent>) {
+    if (!selected) {
+      setActiveTab((e.target as HTMLLIElement).getAttribute('data-tab-id'));
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (selected) {
+      return;
+    }
+
+    const enabledIndexes = childrenArray.reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    const enabledCurrentIndex = enabledIndexes.indexOf(currentIndex);
+
+    let index: number;
+
+    switch (e.key) {
+      case 'ArrowRight':
```

Comment:
We want arrow up and arrow down to have the same behavior? Doesn't seem that's according to [the a11y guidelines](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html).

Reply:
Removed!